### PR TITLE
doc: GraphQL Generation topic

### PR DESCRIPTION
**Pull request by moxley**
Add a topic for GraphQL Generation

### Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Features include unit/acceptance tests
- [x] Documentation

### CODE REVIEW

File: documentation/topics/graphql-generation.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+    queries do
+      # create a field called `list_tickets` that uses the `read` read action to fetch a list of tickets
+      list :list_tickets, :read 
+    end
+  end
+```
+
+The second argument to `list :list_tickets, :read` is the action that will be called when the query is
+run. In the current example, the action is `:read`, which is the generic Read action. We will need
```
Comment (zachdaniel) [inline documentation/topics/graphql-generation.md:127]:
```suggestion
run. In the current example, the action is `:read`, which is the generic Read action. Lets create
a custom action in order to define query parameters for the `list_tickets` query.
```

File: documentation/topics/graphql-generation.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    queries do
+      # create a field called `list_tickets` that uses the `read` read action to fetch a list of tickets
+      list :list_tickets, :read 
+    end
+  end
+```
+
+The second argument to `list :list_tickets, :read` is the action that will be called when the query is
+run. In the current example, the action is `:read`, which is the generic Read action. We will need 
+to create a custom action in order to define query parameters for the `list_tickets` query.
```
Comment (zachdaniel) [inline documentation/topics/graphql-generation.md:128]:
```suggestion
run. In the current example, the action is `:read`, which is the generic Read action. Lets create 
a custom action in order to define query parameters for the `list_tickets` query.
```

You can technically take over the primary read, i.e
```elixir
defaults [:update, :destroy]

read :read do
  primary? true
  argument ...
end
```

This just isn't typically a good idea.

File: documentation/topics/graphql-generation.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  }
+}
+```
+
+## More GraphQL Docs
+
+If you haven't already, please turn on the documentation tag for AshGraphql. Tags can be controlled
+at the top of the left navigation menu, under "Including Libraries:".
+
+- [Getting Started With GraphQL](/docs/guides/ash_graphql/latest/tutorials/getting-started-with-graphql)
```
Comment (zachdaniel) [inline documentation/topics/graphql-generation.md:176]: So the way that we do links in docs is a bit interesting. We have some tools that transform links, but in this case we'd do it like this
```suggestion
- [Getting Started With GraphQL](/documentation/tutorials/getting-started-with-graphql)
```

File: documentation/topics/graphql-generation.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+- [AshGraphql API](/docs/dsl/ash_graphql/latest/ashgraphql-api)
```
Comment (zachdaniel) [inline documentation/topics/graphql-generation.md:177]:
```suggestion
- `AshGraphql.Api`
```
The above will be replaced in hexdocs and ash_hq, although it won't be a link in github (which is the least important of the three targets we have for docs)

File: documentation/topics/graphql-generation.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
+# GraphQL Query Generation
+
+## GraphQL Request and Response
+
+Following where we left off from [Getting Started with GraphQL](/docs/guides/ash_graphql/latest/tutorials/getting-started-with-graphql), we'll explore what the GraphQL
```
Comment (zachdaniel) [inline documentation/topics/graphql-generation.md:5]:
```suggestion
Following where we left off from [Getting Started with GraphQL](/documentation/tutorials/getting-started-with-graphql.md), we'll explore what the GraphQL
```

**Comment by zachdaniel** [comment]
This looks great! Would you mind also adding a link to this guide from the bottom of the ash_graphql getting started guide? I've shown some examples of how we do links for compatibility across the various places we show docs.

**Comment by moxley** [comment]
@zachdaniel Those are great suggestions, and I added them to the PR.

File: documentation/tutorials/getting-started-with-graphql.md [hunk 6]
```diff
@@ -149,3 +149,16 @@ end
 If you started with `mix new ...` instead of `mix phx.new ...` and you want to
 still use phoenix, the fastest path that way is typically to just create a new
 phoenix application and copy your resources/config over.
+
+## What's next?
```
Comment (zachdaniel) [inline documentation/tutorials/getting-started-with-graphql.md:153]: This is great, I didn't realize we didn't even have a section for what to look at next. Thank you for adding this and adding more links! Great work.

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**